### PR TITLE
Component/companies/id

### PR DESCRIPTION
**Pull request by Chabelle78**
Component details of project is rerendering --> infos contacts cannot be updated.

### CODE REVIEW

File: src/components/Routes.tsx [hunk 1]
```diff
@@ -1,17 +1,31 @@
...
 
 function Routes(): JSX.Element {
   return (
     <>
       <Route exact path="/aeviso" component={SuperAdmin} />
       <Route exact path="/home" component={HomePage} />
       <Route exact path="/clients" component={ListsCompanies} />
+      <Route exact path="/companies" component={Companies} />
+      <Route exact path="/companies/:id" component={Company} />
+      <Route exact path="/companies/*/projects/:id" component={DetailsProjects} />
```
Comment (Karnak19) [inline src/components/Routes.tsx]: `*` ?

Explain me pls

File: src/components/company/DetailsProjects.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { useQuery } from 'react-query';
+import { useParams } from 'react-router-dom';
+import { AxiosError } from 'axios';
+import { project } from '../../API/requests';
+import Spinner from '../Spinner';
+import InformationsCompany from './InformationsCompany';
+
+function DetailsProjects(): JSX.Element {
+  const { id }: { id: string } = useParams();
```
Comment (Karnak19) [inline src/components/company/DetailsProjects.tsx]: Not a big deal but

```js
  const { id } = useParams<{ id: string }>();
```

File: src/components/company/InformationsCompany.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { useQuery } from 'react-query';
+import { useParams } from 'react-router-dom';
+import { AxiosError } from 'axios';
+import { companies } from '../../API/requests';
+import Spinner from '../Spinner';
+
+function InformationsCompany(): JSX.Element {
+  const { id }: { id: string } = useParams();
```
Comment (Karnak19) [inline src/components/company/InformationsCompany.tsx]: Here too

```js
  const { id } = useParams<{ id: string }>();
```

File: src/components/company/ProjectsCieAdmin.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import { AxiosError } from 'axios';
+import { companies } from '../../API/requests';
+import { useForm } from 'react-hook-form';
+import { Link } from 'react-router-dom';
+
+import Spinner from '../Spinner';
+import SearchInput from '../SearchInput';
+
+function ProjectsCieAdmin(): JSX.Element {
+  const { id }: { id: string } = useParams();
```
Comment (Karnak19) [inline src/components/company/ProjectsCieAdmin.tsx]: Also here

```js
const { id } = useParams<{ id: string }>();
```

File: src/components/Routes.tsx (hunk 1 above)
Comment (Aeryle) [reply src/components/Routes.tsx]: ~~The `*` here works the same way as on Express, we don't need the company's id so we used this instead~~

Comment (Karnak19) [reply src/components/Routes.tsx]: So why not just `/project/:id` ?

Comment (Aeryle) [reply src/components/Routes.tsx]: There has been a problem of logic on the way, this will be fixed

Comment (Chabelle78) [reply src/components/Routes.tsx]: fixed !

**Pull request closed without merging**